cmd/cloud.go

```diff
@@ -235,10 +236,18 @@ This will execute the test on the Load Impact cloud service. Use "k6 login cloud
 	},
 }
 
+func cloudCmdFlagSet() *pflag.FlagSet {
+	flags := pflag.NewFlagSet("", pflag.ContinueOnError)
+	flags.SortFlags = false
+	flags.AddFlagSet(optionFlagSet())
+	flags.AddFlagSet(runtimeOptionFlagSet(false))
+	//TODO: figure out a better way to handle the CLI flags - global variables are not very testable... :/
```

Comment:
Well you can use local variables but you will need to not use `init` :) . I suppose at some point we will rewrite it but this seems as too small of a problem atm

Reply:
It likely needs to be rewritten as we tackle [the big config refactoring](https://github.com/loadimpact/k6/issues/883)... Using global variables is somewhat fine, until you need to test them. And it can hide issues like the fact we currently somehow have 2 different variables for saving the value of the `--config` flag ([details](https://github.com/loadimpact/k6/issues/883#issuecomment-468646291)).

But in my head, we have at least 2 other similar issues that are more problematic than this, both in terms of improving test coverage and correctness, and in their potential for hiding bugs...

The first problem is the liberal use of `os.GetEnv()` to read environment variables directly into those same global variables. `grep -FR --exclude-dir=vendor "os.Getenv"` produces
```
./cmd/config.go:var configFile = os.Getenv("K6_CONFIG") // overridden by `-c` flag!
./cmd/run.go:	runType       = os.Getenv("K6_TYPE")
./cmd/run.go:	runNoSetup    = os.Getenv("K6_NO_SETUP") != ""
./cmd/run.go:	runNoTeardown = os.Getenv("K6_NO_TEARDOWN") != ""
./cmd/cloud.go:	exitOnRunning = os.Getenv("K6_EXIT_ON_RUNNING") != ""
```
That's on top of the other issues we also have due to using `envconfig`... Ideally, we shouldn't directly call `os.Getenv()` or other `os` functions more than once. Instead, we should just capture the environment variables once per run, in a single place in the code, and then pass them through to the configuration parsing and setting logic. That way we can actually mock the process and write simple and sensible tests that are able to check all of the configuration parsing and management logic... 

The second problem lies in the way we're dealing with the CLI flags that don't save their values into global variables, i.e. the majority of the CLI flags. And I'm not even talking about how we use them to store [both the default option values and the highest-priority user-supplied ones](https://github.com/loadimpact/k6/blob/0b1b25949256544226cbe3641ad8024c06ef3420/cmd/config.go#L202-L206)... The problem I've hit a few times recently is that a lot of the potential developer errors in setting CLI flags won't happen at compile time, but rather at runtime. 

The problematic code can be seen [here](https://github.com/loadimpact/k6/blob/0b1b25949256544226cbe3641ad8024c06ef3420/cmd/options.go#L77-L92) and [here](https://github.com/loadimpact/k6/blob/0b1b25949256544226cbe3641ad8024c06ef3420/cmd/config.go#L122-L125) (and in a few other places as well). Because we're using string identifiers in multiple places, and we [`panic()` if we don't have the expected flag](https://github.com/loadimpact/k6/blob/0b1b25949256544226cbe3641ad8024c06ef3420/cmd/common.go#L71-L101), and we don't have tests for the different `cobra.Command` global variables ( i.e. `runCmd`, `cloudCmd`, etc.), minor developer errors can lead to releasing broken code, especially in rarely used commands...

So yeah, I won't fix this specific TODO with this pull request, but it and the others I mentioned above should definitely be fixed sometime soon...